mono/mini/interp/interp.c

```diff
@@ -173,8 +173,8 @@ debug_enter (MonoInvocation *frame, int *tracing)
 		g_print  ("%s)\n", args);
 		g_free (args);
 	}
-	if (mono_profiler_events & MONO_PROFILE_ENTER_LEAVE)
-		mono_profiler_method_enter (frame->runtime_method->method);
+	if (mono_profiler_should_instrument_method (frame->runtime_method->method, TRUE))
```

Comment:
Does that really add any value? "Enabling" enter/leave, in the new API, means installing a filter function that gets called here. If no filter function is installed, no enter/leave instrumentation is done.